pagopa/io-app
# feat(Firma con IO): [SFEQS-1892] Add environment to Mixpanel events

### This PR depends on #5179 

Co-authored by @hevelius 

## Short description
This PR updates the FCI definitions and incorporates an environment property within FCI related analytics events. The environment flag is extracted from the `env` header in the `getSignatureDetailViewBy` API call. This flag is integrated into every Mixpanel tracking function by extracting it with a selector from the state, retrieved using the `store.getState()` function. This is necessary as direct access to the store is unavailable in the `analytics.ts` file without the appropriate context.

## List of changes proposed in this pull request
- Updates FCI definitions;
- Updates analytics function to use the `buildEventProperties` utility; 
- Adds a state section to store the environment with a dedicated selector;
- Adds `FciHeaders` decoder to fetch the unknown `headers` in the `getSignatureDetailViewBy` API call as it doesn't get properly typed from the auto generated client. This decoder is then used in `handleGetSignatureRequestById.ts` to fill the aformentioned store section with the `env` value. 


## How to test
Start a signing flow using either the production environment file or a local setup with the io-dev-api-server by setting the `waitForSignatureCount: 1` flag. Use console.log or a tool like Flipper or reactotron to check the environment state under `features->fci->env`.
Optionally, execute a test in the production environment and verify that the mixpanel console accurately receives the expected data.
<img width="1330" alt="Screenshot 2023-11-07 at 16 16 28" src="https://github.com/pagopa/io-app/assets/49342144/921a0081-5bca-40b0-8cee-4c65b84a3055">
